hab/utils/training_helper.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import torch
+from torch.nn import Module
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from hab.utils import habs_logging
+
+# ------------ logging ------------
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s HABs:%(levelname)s - %(name)s - %(message)s"
+)
+
+logging.captureWarnings(True)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(habs_logging.ch)
+logger.addHandler(habs_logging.fh)
+# ---------------------------------
+
+
+def training_laps(
+        model: Module,
+        data_loader: DataLoader,
+        epochs: int,
+        optimizer: Optimizer,
+        criterion: Module
+):
+    """
+    Execute training on model by completing specified number of epochs/laps through database.
+
+    :param model:Model to be trained.
+    :param data_loader: Data loader that contains training data.
+    :param epochs: Number of laps to complete through dataset.
+    :param optimizer: Optimization algorithm used to train the model.
+    :param criterion: Loss function used to train the model.
+    """
+    for epoch in range(epochs):
```

Comment:
Since you call `model.eval()` in the eval function, you need to call `model.train()` before your training loop starts. Otherwise none of the regularization will be present during training.

Reply:
Ok thanks. I called `model.eval()` in my new `validation_lap` function and `model.train()` in the `training_lap` function. Is that what you were talking about?

Reply:
During our call on 5.5.21 Ty confirmed this was correct.